src/sxg_sig.c

```diff
@@ -22,71 +22,116 @@
-  sxg_buffer_release(&target->sig);
-  sxg_buffer_release(&target->validity_url);
+  OPENSSL_free(target->integrity);
+  target->integrity = NULL;
+  OPENSSL_free(target->sig);
+  target->sig = NULL;
+  OPENSSL_free(target->validity_url);
+  target->validity_url = NULL;
 }
 
 bool sxg_sig_set_name(const char* name, sxg_sig_t* sig) {
-  sxg_buffer_release(&sig->name);
-  return sxg_write_string(name, &sig->name);
+  sig->name_size = strlen(name);
+  sig->name = OPENSSL_realloc(sig->name, sig->name_size + 1);
+  if (sig->name == NULL) {
+    return false;
+  }
+  strcpy(sig->name, name);
+  return true;
 }
 
 bool sxg_sig_set_cert_sha256(X509* certificate, sxg_sig_t* sig) {
-  sxg_buffer_release(&sig->cert_sha256);
-  return sxg_calculate_cert_sha256(certificate, &sig->cert_sha256);
+  sig->cert_sha256_size = SHA256_DIGEST_LENGTH;
+  sig->cert_sha256 = OPENSSL_realloc(sig->cert_sha256, sig->cert_sha256_size);
+  if (sig->cert_sha256 == NULL) {
+    return false;
+  }
+  return sxg_calculate_cert_sha256(certificate, sig->cert_sha256);
 }
 
 bool sxg_sig_set_cert_url(const char* cert_url, sxg_sig_t* sig) {
-  sxg_buffer_release(&sig->cert_url);
-  return sxg_write_string(cert_url, &sig->cert_url);
+  sig->cert_url_size = strlen(cert_url);
+  sig->cert_url = OPENSSL_realloc(sig->cert_url, sig->cert_url_size + 1);
+  if (sig->cert_url == NULL) {
+    return false;
+  }
+  strcpy(sig->cert_url, cert_url);
+  return true;
 }
 
 bool sxg_sig_set_ed25519key(const EVP_PKEY* public_key, sxg_sig_t* sig) {
-  sxg_buffer_release(&sig->ed25519key);
   size_t key_len;
-  return (EVP_PKEY_get_raw_public_key(public_key, NULL, &key_len) == 1) &&
-         sxg_buffer_resize(key_len, &sig->ed25519key) &&
-         (EVP_PKEY_get_raw_public_key(public_key, sig->ed25519key.data,
-                                      &key_len) == 1);
+  if (EVP_PKEY_get_raw_public_key(public_key, NULL, &key_len) != 1) {
+    return false;
+  }
+  sig->ed25519key = OPENSSL_realloc(sig->ed25519key, key_len);
+  if (sig->ed25519key == NULL) {
+    return false;
+  }
+  return EVP_PKEY_get_raw_public_key(public_key, sig->ed25519key, &key_len) ==
+         1;
 }
 
 bool sxg_sig_set_integrity(const char* integrity, sxg_sig_t* sig) {
-  sxg_buffer_release(&sig->integrity);
-  return sxg_write_string(integrity, &sig->integrity);
+  sig->integrity_size = strlen(integrity);
+  sig->integrity = OPENSSL_realloc(sig->integrity, sig->integrity_size + 1);
+  if (sig->integrity == NULL) {
+    return false;
+  }
+  strcpy(sig->integrity, integrity);
+  return true;
 }
 
 bool sxg_sig_set_validity_url(const char* validity_url, sxg_sig_t* sig) {
-  sxg_buffer_release(&sig->validity_url);
-  return sxg_write_string(validity_url, &sig->validity_url);
+  sig->validity_url_size = strlen(validity_url);
+  sig->validity_url =
+      OPENSSL_realloc(sig->validity_url, sig->validity_url_size + 1);
+  if (sig->validity_url == NULL) {
+    return false;
+  }
+  strcpy(sig->validity_url, validity_url);
```

Comment:
`strcpy` is dangerous and flagged by clang-tidy.  Does `strncpy` or `memcpy` make more sense here?

Reply:
replaced with strncpy.